### PR TITLE
SLM Twig UI

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Based on #16910.

- Migrate UI code for SLM/SLA/OLA related code to Twig
- A fair amount of de-duplication of UI code

UX change for escalation levels list. I'm not sure if it changed with a previous Twig PR since it reuses UI code from rules, or was always like this, but the escalation levels list looked too crowded and wasn't easy for me (someone who doesn't use service levels) to understand what was going on.

Before (Without any criteria or actions):
Selection_291

After (With a criteria and action):
Selection_292

The change makes the list cleaner but also means that the criteria and actions cannot be added/changed from the escalation level list anymore. You have to click the escalation level link to open its form.

### CODE REVIEW

**Review by orthagh — CHANGES_REQUESTED** [review]
Some issues may be in the main branch directly. I didn't have the time to investigate more.

- form for adding an escalation level seems to lose the relation with the parent SLA/OLA
- I got an incoherence between the number of tickets counted in the tab (32) vs the tab content (no data). 
- general advice: if you find a number dropdown without any special option (like the one for maximum time), replace it with a simple `input[type=number]`

Regarding the UI/UX, nothing to say

**Comment by cconard96** [comment]
>     * form for adding an escalation level seems to lose the relation with the parent SLA/OLA

Fixed
 
>     * I got an incoherence between the number of tickets counted in the tab (32) vs the tab content (no data).

Doesn't seem related. This tab also triggers a deprecation warning. It looks like the way this tab is handled was changed a few times in the last year or two.
This tab is still using raw HTML, so I'll probably address the issue it at some point in this project.

**Review by orthagh — APPROVED** [review]

File: src/LevelAgreement.php [hunk 1]
```diff
@@ -584,7 +515,8 @@ public static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $
      *
      * @param $tickets_id
      * @param $type
-     * @return false|iterable
+     * @return false|Generator
```
Comment (cedric-anne) [inline src/LevelAgreement.php:518]:
```suggestion
     * @return false|iterable
```

File: src/LevelAgreementLevel.php [hunk 2]
```diff
@@ -414,10 +394,148 @@ public static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $
...
+                'canedit' => $canedit,
+            ],
+        ]);
+    }
+
+    /**
+     * @param LevelAgreement $la The Level Agreement object (SLA or OLA)
+     * @return void
+     */
+    public function showForLA(LevelAgreement $la): void
```
Comment (cedric-anne) [inline src/LevelAgreementLevel.php:438]: It is used only by subclasses, and I do not see any reason to permit this method to be overriden. We can make it final and protected.

```suggestion
    final protected function showForLA(LevelAgreement $la): void
```

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**